src/xcode/ENA/ENA/Source/AppDelegate+PlausibleDeniability.swift

```diff
@@ -65,17 +67,19 @@ extension AppDelegate {
 		}
 	}
 
-	func executeFakeRequestOnAppLaunch() {
-		// Execute a fake request 1 in 100 times while we are running in foreground.
-		// We therefore define a magic number and only send a fake request when
-		// our random number from [1, 100] matches the magic number.
-		let magicNumber = 6
-		guard
-			UIApplication.shared.applicationState == .active,
-			Int.random(in: 1...100) == magicNumber
-		else { return }
 
-		sendFakeRequest()
+	/// Randomly execute a fake request
+	/// - Parameter propability: the probability p to execute a fake request. Accepting values between 0 and 1.
+	/// - Returns: Bool to indicate wether the fake request has been sent
+	@discardableResult
+	func executeFakeRequestOnAppLaunch(propability p: Double) -> Bool {
+		assert(p <= 1, "p should be lighter than or equal 1.0")
+		assert(p >= 0, "p should be greater than or equal 0.0")
+		if Double.random(in: 0...1) <= p {
```

Comment:
If I'm not mistaken, this can still be executed even if p is set to 0.0, because Double.random(in: 0...1) can return 0

Reply:
I was wondering the same but it seems the bounds seem not to be reached for `p = 0.0`. Due to time and other constraints, I haven't added a unit test (yet) but here is my local test for this implementation: 

```swift
// Will never trigger, even for 100M iteration. 
// But will work on very small values like `let p = 0.0.nextUp` (5e-324)
let p = 0.0

let total: Int = 1_000_000
var hits: Int = 0

for _ in 0...total {
    if Double.random(in: 0...1) <= p {
        hits += 1
    }
}

print("\(hits)/\(total) => \(Double(hits)/Double(total))")
```

Reply:
I would consider a compiler 'optimization' here and would expect hits with `NSDecimalNumbers` instead of `Double`.

To short cut the discussion, we could go with:
```suggestion
		if Double.random(in: 0.0.nextUp...1) <= p {
```

Reply:
The nextUp solution looks good to me, just to be safe 👍